watertap/property_models/unit_specific/activated_sludge/asm3_reactions.py

```diff
@@ -0,0 +1,1013 @@
+        add_object_reference(self, "i_SSSTO", self.config.property_package.i_SSSTO)
+
+        # Kinetic Parameters
+        k_H_dict = {"10C": 2, "20C": 3}
+        self.k_H = pyo.Var(
+            k_H_dict.keys(),
+            domain=pyo.PositiveReals,
+            initialize=k_H_dict,
+            units=pyo.units.day**-1,
+            doc="Hydrolysis rate constant (g-COD-X_S / g-COD-X_H / day)",
+        )
+        self.K_X = pyo.Var(
+            initialize=1,
+            units=pyo.units.dimensionless,
+            domain=pyo.PositiveReals,
+            doc="Hydrolysis saturation constant (g-COD-X_S / g-COD-X_H)",
+        )
+
+        # Heterotrophic organisms X_H, aerobic and denitrifying activity
+        k_STO_dict = {"10C": 2.5, "20C": 5}
+        self.k_STO = pyo.Var(
+            k_STO_dict.keys(),
+            domain=pyo.PositiveReals,
+            initialize=k_STO_dict,
+            units=pyo.units.day**-1,
+            doc="Storage rate constant (g-COD-S_S / g-COD-X_H / day)",
+        )
+        self.eta_NOX = pyo.Var(
+            initialize=0.6,
+            units=pyo.units.dimensionless,
+            domain=pyo.PositiveReals,
+            doc="Anoxic reduction factor",
+        )
+        self.K_O2 = pyo.Var(
+            initialize=0.2e-3,
+            units=pyo.units.kg / pyo.units.m**3,
+            domain=pyo.PositiveReals,
+            doc="Saturation constant for S_NO2 (kg-O2 / m3)",
+        )
+        self.K_NOX = pyo.Var(
+            initialize=0.5e-3,
+            units=pyo.units.kg / pyo.units.m**3,
+            domain=pyo.PositiveReals,
+            doc="Saturation constant for S_NOX (kg-N-NO3 / m3)",
+        )
+        self.K_S = pyo.Var(
+            initialize=2e-3,
+            units=pyo.units.kg / pyo.units.m**3,
+            domain=pyo.PositiveReals,
+            doc="Saturation constant for substrate S_S (kg-COD-S_S / m3)",
+        )
+        self.K_STO = pyo.Var(
+            initialize=1,
+            units=pyo.units.dimensionless,
+            domain=pyo.PositiveReals,
+            doc="Saturation constant for for X_STO (g-COD-X_STO / g-COD-X_H)",
+        )
+        mu_H_dict = {"10C": 1, "20C": 2}
+        self.mu_H = pyo.Var(
+            mu_H_dict.keys(),
+            domain=pyo.PositiveReals,
+            initialize=mu_H_dict,
+            units=pyo.units.day**-1,
+            doc="Heterotrophic max. growth rate of X_H (day^-1)",
+        )
+        self.K_NH4 = pyo.Var(
+            initialize=0.01e-3,
+            units=pyo.units.kg / pyo.units.m**3,
+            domain=pyo.PositiveReals,
+            doc="Saturation constant for ammonium, S_NH4 (kg-N / m3)",
+        )
+        self.K_ALK = pyo.Var(
+            initialize=0.1e-3,
+            units=pyo.units.kmol / pyo.units.m**3,
+            domain=pyo.PositiveReals,
+            doc="Saturation constant for alkalinity for X_H (kmol-HCO3- / m3)",
+        )
+        b_H_O2_dict = {"10C": 0.1, "20C": 0.2}
```

Comment:
Rather than hard-code "10C" and "20C", I think we should index by "ref_temp_1" and "ref_temp_2". That way, if one wanted to changed the two ref temperatures to something other than 10 and 20C, they could do so.